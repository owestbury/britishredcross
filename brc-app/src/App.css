
.App {
    min-height: 100vh;
    text-align: center;
    text-align: center;
}

.App-header {
    background-color: #282c34;
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
    min-height: 50%;
    padding-bottom: 10px;
}


.App-body {
    width: 60%;
    margin: 20px auto;
}

.App-logo {
    height: 40vmin;
    pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
    .App-logo {
        animation: App-logo-spin infinite 20s linear;
    }
}

.App-link {
    color: #61dafb;
}

@keyframes App-logo-spin {
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
}

.currency {
    padding: 10px;
}

.rt-expandable {
    cursor: unset !important;
}

.show-pointer {
    cursor: pointer !important;
}

.selectedPlans {
    justify-content: center;
}

.btn-group.btn-group-toggle {
    padding: 10px;
}

.plansBox {
    padding: 20px;
    margin: 20px;
    -webkit-border-radius: 3px;
    -moz-border-radius: 3px;
    border-radius: 3px;
    background: #fff;
    -webkit-box-shadow: 0 0 4px 0 rgb(0 0 0 / 32%);
    box-shadow: 0 0 4px 0 rgb(0 0 0 / 32%);
    -ms-border-radius: 3px;
}

.selectedPlansTitle {
    width: 100%;
    padding: 20px;
    color: #111;
    font-weight: bold;
    letter-spacing: -1px;
    line-height: 1;
    text-align: left;
}

.selected {
    padding: 10px;
    justify-content: center;
}

.selected img.card-img-top {
    display: flex;
    justify-content: center;
    width: 150px;
}

