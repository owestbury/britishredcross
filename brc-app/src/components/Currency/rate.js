export const currencyRate = [
    {'code': 'gb', 'symbol': '£', 'rate': 1},
    {'code': 'eu', 'symbol': '€', 'rate': 1.17},
    {'code': 'us', 'symbol': '$', 'rate': 1.37},
    {'code': 'jp', 'symbol': '¥', 'rate': 150.11},
];
