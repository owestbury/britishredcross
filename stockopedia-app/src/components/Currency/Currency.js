import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Dropdown} from 'react-bootstrap';

export class Currency extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    render() {
        const {
            variant,
            onChange,
        } = this.props;

        return (
            <Dropdown>
                <Dropdown.Toggle variant="success" id="dropdown-basic">
                    {variant.title}
                </Dropdown.Toggle>

                <Dropdown.Menu>
                    <Dropdown.Item id="gb" onClick={onChange}>(£) GBP</Dropdown.Item>
                    <Dropdown.Item id="eu" onClick={onChange}>(€) EUR</Dropdown.Item>
                    <Dropdown.Item id="us" onClick={onChange}>($) USD</Dropdown.Item>
                    <Dropdown.Item id="jp" onClick={onChange}>(¥) JAP</Dropdown.Item>
                </Dropdown.Menu>
            </Dropdown>
        );
    }
}

Currency.propTypes = {
    variant: PropTypes.shape({
        title: PropTypes.string,
    }),
    onChange: PropTypes.func.isRequired,
};
