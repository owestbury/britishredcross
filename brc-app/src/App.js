import React, {Component} from 'react';
import css from './App.css';
import {ItemCard} from "./components/ItemCard/ItemCard";
import {CardDeck, ToggleButton, ToggleButtonGroup} from 'react-bootstrap';
import {Currency} from "./components/Currency/Currency";
import {currencyRate} from "./components/Currency/rate"
import {BillingCard} from "./components/Billing/Billing"

class App extends Component {

    handleChange = event => {
        this.setState({username: event.target.value});
    };

    constructor(props) {
        super(props);
        this.state = {
            page: {
                title: 'British Red Cross - Available subscription plans',
                subscription: 'Subscription',
                subscription_plan: 'Subscription Plans',
                content_order: 'Contents of the order',
                select: 'Please select any subscription',
                summary_title: 'Summary',
                button: {
                    currency: 'Currency',
                    monthly: 'Monthly',
                    annual: 'Annual',
                    remove: 'Remove',
                    process: 'Go ahead and process it?'
                },
            },
            countries: [],
            subscription: false,
            SubSelected: [],
            val: 'Annual',
            currency: {'code': 'gb', 'symbol': 'Â£', 'rate': 1},
            variant: {
                title: "Currency"
            }
        };
    }

    componentDidMount() {
        this.getCountries();
    }

    getCountries = _ => {
        const path = process.env.REACT_APP_API_COUNTRIES;
        return fetch(path)
            .then(response => response.json())
            .then(response => this.setState({countries: response}))
            .catch(err => console.log(err))
    }

    getLastId = (data) => {
        const pickSubscription = data.slice(-1).pop() ? 'Annual' : 'Monthly';
        return pickSubscription;
    }

    currencySubscription = (currency) => {
        const cur = currency.target.id;
        const getData = currencyRate;
        const res = getData.filter(item => {
            return item.code === cur
        })
        this.setState({currency: res[0]})
    }

    selectSub = (event, newFormats) => {
        const sel = event.slice(-1).pop() ? event.slice(-1).pop() : null;
        if (sel) {
            this.setState({val: event.slice(-1).pop()})
            this.setState({SubSelected: []})
        }
    };

    b = (idToSearch) => {
        return this.state.SubSelected.filter(item => {
            return item.id === idToSearch
        })
    };

    handleSubscription = (index) => {
        const item = index[0];

        if (this.b(item.id).length === 0) {
            this.setState({SubSelected: this.state.SubSelected.concat(item)})
            this.state.SubSelected.pop();
        }
    };

    removeSub = (index) => {
        const item = index[0];
        const myArr = this.state.SubSelected.findIndex(function (o) {
            return o.id === item;
        })
        if (myArr !== -1) this.state.SubSelected.splice(myArr, 1);
        this.setState({SubSelected: this.state.SubSelected});
    }

    render() {
        const {page, countries, SubSelected, val, variant, currency} = this.state;

        return (
            <div className="App">
                <header className="App-header" aria-lable={page.title}>
                    <h2>
                        {page.title}
                    </h2>
                </header>

                <h4 data-testid="subscription" aria-lable={page.subscription}>{page.subscription}</h4>

                <div className="currency">

                    <Currency
                        variant={variant}
                        onChange={(index) => this.currencySubscription(index)}
                    />

                    <div>
                        <pre><strong>Rate:</strong> {currency.code} | {currency.symbol}{currency.rate}</pre>
                    </div>
                </div>
                <ToggleButtonGroup type="checkbox" value={val} onChange={this.selectSub}>
                    <ToggleButton value="Monthly" aria-label="Monthly">Monthly</ToggleButton>
                    <ToggleButton value="Annual" aria-label="Annual">Annual</ToggleButton>
                </ToggleButtonGroup>

                <div className={css.cardDeck}>
                    <div className="selectedPlansTitle" aria-label={page.subscription_plan}>
                        {page.subscription_plan}
                    </div>
                    <CardDeck>
                        {
                            countries.map((item, index) =>
                                <ItemCard
                                    key={item.id}
                                    index={index}
                                    image={false}
                                    item={item}
                                    onChange={(index) => this.handleSubscription(index)}
                                    subscription={1}
                                    val={val}
                                    currency={currency}
                                    button={page.button}
                                />
                            )
                        }
                    </CardDeck>
                </div>
                <div className="plansbox">
                    {this.state.SubSelected.length > 0 ?
                        <div className="selectedPlans">
                            <div className="selectedPlansTitle" aria-label={page.content_order}>
                                {page.content_order}
                            </div>
                            <div className="selected">
                                <CardDeck>
                                    {
                                        SubSelected.map((item, index) =>
                                            <ItemCard
                                                key={item.id}
                                                index={index}
                                                image={false}
                                                item={item}
                                                onChange={(index) => this.removeSub(index)}
                                                subscription={2}
                                                val={val}
                                                confirm={SubSelected ? this.getLastId(SubSelected) : null}
                                                currency={currency}
                                                button={page.button}
                                            />
                                        )
                                    }
                                </CardDeck>
                            </div>
                        </div> :
                        <div aria-label={page.select}>
                            {page.select}
                        </div>
                    }
                    {this.state.SubSelected.length > 0 ?
                        <div className="selected-total">
                            <div className={page.summary_title}>
                                {page.summary_title}
                            </div>
                            <CardDeck>
                                {
                                    <BillingCard
                                        index={SubSelected}
                                        image={false}
                                        item={SubSelected}
                                        onChange={(index) => this.removeSub(index)}
                                        subscription={2}
                                        val={val}
                                        confirm={SubSelected ? this.getLastId(SubSelected) : null}
                                        currency={currency}
                                        button={page.button}
                                    />
                                }
                            </CardDeck>
                        </div> :
                        <div>
                        </div>
                    }
                </div>
            </div>
        );
    }
}

export default App;
