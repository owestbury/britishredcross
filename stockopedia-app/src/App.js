import React, {Component} from 'react';
import css from './App.css';
import {ItemCard} from "./components/ItemCard/ItemCard";
import {CardDeck, ToggleButton, ToggleButtonGroup} from 'react-bootstrap';
import {Currency} from "./components/Currency/Currency";
import {currencyRate} from "./components/Currency/rate"
import {BillingCard} from "./components/Billing/Billing"

class App extends Component {

    handleChange = event => {
        this.setState({username: event.target.value});
    };

    constructor(props) {
        super(props);
        this.state = {
            countries: [],
            subscription: false,
            SubSelected: [],
            val: 'Annual',
            currency: {'code': 'gb', 'symbol': 'Â£', 'rate': 1},
            variant: {
                title: "Currency"
            }
        };
    }

    componentDidMount() {
        this.getCountries();
    }

    getCountries = _ => {
        const path = 'http://localhost:8080/api/countries';
        fetch(path)
            .then(response => response.json())
            .then(response => this.setState({countries: response}))
            .catch(err => console.log(err))

        console.log(this.state);
    }

    getLastId = (data) => {
        const pickSubscription = data.slice(-1).pop() ? 'Annual' : 'Monthly';
        return pickSubscription;
    }

    currencySubscription = (currency) => {
        const cur = currency.target.id;
        const getData = currencyRate;
        const res = getData.filter(item => {
            return item.code === cur
        })
        this.setState({currency: res[0]})
    }

    selectSub = (event, newFormats) => {
        this.setState({val: event.slice(-1).pop()})
        this.setState({SubSelected: []})
    };

    b = (idToSearch) => {
        return this.state.SubSelected.filter(item => {
            return item.id === idToSearch
        })
    };

    handleSubscription = (index) => {
        const item = index[0];

        if (this.b(item.id).length === 0) {
            this.setState({SubSelected: this.state.SubSelected.concat(item)})
            this.state.SubSelected.pop();
        }
    };

    removeSub = (index) => {
        const item = index[0];
        const myArr = this.state.SubSelected.findIndex(function (o) {
            return o.id === item;
        })
        if (myArr !== -1) this.state.SubSelected.splice(myArr, 1);
        this.setState({SubSelected: this.state.SubSelected});
    }

    render() {
        const {countries, SubSelected, val, variant, currency} = this.state;

        return (
            <div className="App">
                <header className="App-header">
                    <h2>
                        Stockopedia - Available subscription plans
                    </h2>
                </header>

                <h4 data-testid="subscription">Subscription</h4>

                <div className="currency">

                    <Currency
                        variant={variant}
                        onChange={(index) => this.currencySubscription(index)}
                    />

                    <div>
                        <pre><strong>Rate:</strong> {currency.code} | {currency.symbol}{currency.rate}</pre>
                    </div>
                </div>
                <ToggleButtonGroup type="checkbox" value={val} onChange={this.selectSub}>
                    <ToggleButton value="Monthly" aria-label="Monthly">Monthly</ToggleButton>
                    <ToggleButton value="Annual" aria-label="Annual">Annual</ToggleButton>
                </ToggleButtonGroup>

                <div className={css.cardDeck}>
                    <CardDeck>
                        {
                            countries.map((item, index) =>
                                <ItemCard
                                    key={item.id}
                                    index={index}
                                    image={false}
                                    item={item}
                                    onChange={(index) => this.handleSubscription(index)}
                                    subscription={1}
                                    val={val}
                                    currency={currency}
                                />
                            )
                        }
                    </CardDeck>
                </div>
                <div className="plansbox">
                    {this.state.SubSelected.length > 0 ?
                        <div className="selectedPlans">
                            <div className="selectedPlansTitle">
                                Subscription Plans
                            </div>
                            <div className="selected">
                                <CardDeck>
                                    {
                                        SubSelected.map((item, index) =>
                                            <ItemCard
                                                key={item.id}
                                                index={index}
                                                image={false}
                                                item={item}
                                                onChange={(index) => this.removeSub(index)}
                                                subscription={2}
                                                val={val}
                                                confirm={SubSelected ? this.getLastId(SubSelected) : null}
                                                currency={currency}
                                            />
                                        )
                                    }
                                </CardDeck>
                            </div>
                        </div> :
                        <div>
                            Please select any subscription
                        </div>
                    }
                    {this.state.SubSelected.length > 0 ?
                        <div className="selected-total">
                            <CardDeck>
                                {
                                    <BillingCard
                                        index={SubSelected}
                                        image={false}
                                        item={SubSelected}
                                        onChange={(index) => this.removeSub(index)}
                                        subscription={2}
                                        val={val}
                                        confirm={SubSelected ? this.getLastId(SubSelected) : null}
                                        currency={currency}
                                    />
                                }
                            </CardDeck>
                        </div> :
                        <div>
                        </div>
                    }
                </div>
            </div>
        );
    }
}

export default App;
