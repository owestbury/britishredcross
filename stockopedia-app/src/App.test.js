import React from 'react';
import App from './App';
import {Currency} from './components/Currency/Currency';
import { render, screen, cleanup, fireEvent } from "./testing/custom-render";

window.fetch = jest.fn(() => {
  const currency = { 'code': 'gb', 'symbol': 'Â£', 'rate': 1 };

  return Promise.resolve({
    json: () => Promise.resolve(currency),
  });
});

const jsdomScroll = window.scroll;
beforeAll(() => {
  // Mock window.scroll - otherwise, Jest/JSDOM will print a not-implemented error.
  window.scroll = () => {};
});

afterAll(() => {
  window.scroll = jsdomScroll;
});

afterEach(cleanup)

describe("<App /> headline title", () => {
  it("Renders <App /> component correctly", () => {
    render(<App />);
    expect(
        screen.getByText(/Stockopedia - Available subscription plans/i)
    ).toBeInTheDocument();
  });
});

describe("<App /> Match h4 title", () => {
  it('should equal to 0', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('subscription')).toHaveTextContent('Subscription')
  });
});

