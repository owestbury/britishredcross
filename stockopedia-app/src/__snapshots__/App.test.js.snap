// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> 1 should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <h2>
        Stockopedia - Available subscription plans
      </h2>
    </header>
    <h4>
      Subscription
    </h4>
    <div
      class="currency"
    >
      <div
        class="dropdown"
      >
        <button
          aria-expanded="false"
          aria-haspopup="true"
          class="dropdown-toggle btn btn-success"
          id="dropdown-basic"
          type="button"
        >
          Currency
        </button>
      </div>
      <div>
        <pre>
          <strong>
            Rate:
          </strong>
           gb | Â£1
        </pre>
      </div>
    </div>
    <div
      class="btn-group btn-group-toggle"
      role="group"
    >
      <label
        aria-label="Monthly"
        class="btn btn-primary"
      >
        <input
          autocomplete="off"
          type="checkbox"
          value="Monthly"
        />
        Monthly
      </label>
      <label
        aria-label="Annual"
        class="btn active btn-primary"
      >
        <input
          autocomplete="off"
          checked=""
          type="checkbox"
          value="Annual"
        />
        Annual
      </label>
    </div>
    <div>
      <div
        class="card-deck"
      />
    </div>
    <div
      class="plansbox"
    >
      <div>
        Please select any subscription
      </div>
      <div />
    </div>
  </div>
</DocumentFragment>
`;
